metals/src/main/scala/scala/meta/internal/metals/debug/DebugDiscovery.scala

```diff
@@ -273,13 +274,37 @@ class DebugDiscovery(
         if (mains.nonEmpty) {
           findMainToRun(Map(buildTarget -> mains.toList), params)
         } else if (tests.nonEmpty) {
-          Future {
-            val params = new b.DebugSessionParams(singletonList(buildTarget))
-            params.setDataKind(b.TestParamsDataKind.SCALA_TEST_SUITES)
-            params.setData(tests.asJava.toJson)
-            params
+          DebugProvider.envFromFile(workspace, Option(params.envFile)).map {
+            envFromFile =>
+              val env =
+                Option(params.env).toList.flatMap(DebugProvider.createEnvList)
+
+              val jvmOpts =
+                JvmOpts.fromWorkspaceOrEnvForTest(workspace).getOrElse(Nil)
+              val scalaTestSuite = new b.ScalaTestSuites(
+                tests
+                  .map(
+                    new b.ScalaTestSuiteSelection(
+                      _,
+                      Nil.asJava,
+                    )
+                  )
+                  .asJava,
+                Option(params.jvmOptions)
+                  .map(jvmOpts ++ _.asScala)
+                  .getOrElse(jvmOpts)
+                  .asJava,
+                (envFromFile ::: env).asJava,
+              )
+              val debugParams = new b.DebugSessionParams(
+                singletonList(buildTarget)
+              )
+              debugParams.setDataKind(
+                b.TestParamsDataKind.SCALA_TEST_SUITES_SELECTION
+              )
+              debugParams.setData(scalaTestSuite.toJson)
+              debugParams
```

Comment:
I pretty much copy and pasted this from [here](https://github.com/scalameta/metals/blob/main/metals/src/main/scala/scala/meta/internal/metals/debug/DebugProvider.scala#L536) as you suggested. Let me know if you think this bit would be worth factoring out into a separate function.

Reply:
We can refactor it, but it's not 100% neccessary.